crates/daphne-server/src/storage_proxy_connection/kv/mod.rs

```diff
@@ -252,28 +252,74 @@ impl<'h> Kv<'h> {
     }
 
     #[tracing::instrument(
-        name = "kv_put",
+        name = "kv_delete",
         skip_all,
         fields(key, prefix = std::any::type_name::<P>()),
     )]
-    pub async fn put<P>(&self, key: &P::Key, value: P::Value) -> Result<(), Error>
+    async fn put_internal<P>(
+        &self,
+        key: &P::Key,
+        value: P::Value,
+        expiration: Option<u64>,
```

Comment:
For clarity (`task_config.not_after` has type `Time`)
```suggestion
        expiration: Option<Time>,
```